cmake/Modules/Packages/MSCG.cmake

```diff
@@ -19,8 +16,15 @@ if(PKG_MSCG)
       URL https://github.com/uchicago-voth/MSCG-release/archive/1.7.3.1.tar.gz
       URL_MD5 8c45e269ee13f60b303edd7823866a91
       SOURCE_SUBDIR src/CMake
-      CMAKE_ARGS -DCMAKE_INSTALL_PREFIX=<INSTALL_DIR> ${CMAKE_REQUEST_PIC} ${EXTRA_MSCG_OPTS}
-      BUILD_COMMAND make mscg INSTALL_COMMAND ""
+      CMAKE_ARGS ${CMAKE_REQUEST_PIC} ${EXTRA_MSCG_OPTS}
+                 -DCMAKE_C_COMPILER=${CMAKE_C_COMPILER}
+                 -DCMAKE_CXX_COMPILER=${CMAKE_CXX_COMPILER}
+                 -DCMAKE_Fortran_COMPILER=${CMAKE_Fortran_COMPILER}
+                 -DCMAKE_INSTALL_PREFIX=<INSTALL_DIR>
+                 -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE}
+                 -DCMAKE_MAKE_PROGRAM=${CMAKE_MAKE_PROGRAM}
+      BUILD_COMMAND ${CMAKE_MAKE_PROGRAM} libmscg.a INSTALL_COMMAND ""
```

Comment:
I think we should use `${CMAKE_COMMAND} --build . --target libmscg.a` as it is more generic.

Reply:
Ok, @akohlmey that also fixed the build error the CI found.